azurerm/resource_arm_hdinsight.go

```diff
@@ -0,0 +1,888 @@
+	ctx := meta.(*ArmClient).StopContext
+
+	createFuture, err := hdiClusterClient.Create(ctx, resGroup, name, parameters)
+
+	//fmt.Printf("createFuture: %+v\n", createFuture.Future.Response)
+	if err != nil {
+		return err
+	}
+
+	err = createFuture.WaitForCompletion(ctx, hdiClusterClient.Client)
+	if err != nil {
+		return err
+	}
+
+	read, err := hdiClusterClient.Get(ctx, resGroup, name)
+	if err != nil {
+		return err
+	}
+	if read.ID == nil {
+		return fmt.Errorf("Cannot read HDI %s (resource group %s) ID", name, resGroup)
+	}
+
+	d.SetId(*read.ID)
+
+	return resourceArmHDInsightRead(d, meta)
+}
+
+/*
+func resourceArmHDInsightUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).hdiClusterClient
+	ctx := meta.(*ArmClient).StopContext
+
+	id, err := parseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	resGroup := id.ResourceGroup
+	name := id.Path["cluster"]
+
+	// if d.HasChange("app_settings") || d.HasChange("version") {
+	// 	appSettings := expandHDInsightAppSettings(d)
+	// 	settings := web.StringDictionary{
+	// 		Properties: appSettings,
+	// 	}
+	//client.
+	// 	_, err := client.UpdateApplicationSettings(ctx, resGroup, name, settings)
+	// 	if err != nil {
+	// 		return fmt.Errorf("Error updating Application Settings for Function App %q: %+v", name, err)
+	// 	}
+	// }
+
+	return resourceArmHDInsightRead(d, meta)
+
+}
+*/
+func resourceArmHDInsightRead(d *schema.ResourceData, meta interface{}) error {
+	hdinsightClusterClient := meta.(*ArmClient).hdiClusterClient
+	ctx := meta.(*ArmClient).StopContext
+
+	id, err := parseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	resGroup := id.ResourceGroup
+	name := id.Path["clusters"]
+
+	resp, err := hdinsightClusterClient.Get(ctx, resGroup, name)
+
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("Error making Read request on Azure HDInsight %s: %+v", name, err)
+	}
+
+	d.Set("name", resp.Name)
+	d.Set("resource_group_name", resGroup)
+	d.Set("location", azureRMNormalizeLocation(*resp.Location))
+
+	if resp.Properties != nil {
+		if resp.Properties.ClusterVersion != nil {
+			d.Set("cluster_version", resp.Properties.ClusterVersion)
+		}
+		d.Set("tier", resp.Properties.Tier)
+
+		if resp.Properties.ClusterDefinition != nil {
+			d.Set("kind", resp.Properties.ClusterDefinition.Kind)
+		}
+
+		if resp.Properties.SecurityProfile != nil {
+			if err := d.Set("security", schema.NewSet(resourceArmHDInsightSecurityHash, flattenHDInsightSecurityProfile(resp.Properties.SecurityProfile))); err != nil {
+				//	return fmt.Errorf()
+			}
+		}
+
+		if resp.Properties.ComputeProfile != nil && len(*resp.Properties.ComputeProfile.Roles) > 0 {
+			if err := d.Set("role_node", schema.NewSet(resourceArmHDInsightnodeHash, flattenHDInsightRoleNode(resp.Properties.ComputeProfile.Roles))); err != nil {
+				//	return fmt.Errorf()
+			}
+		}
+		// not implemented in azure-sdk
```

Comment:
[This appears to exist under `storage_profile` - although it only allows Storage Accounts at this time?](https://github.com/Azure/azure-sdk-for-go/blob/master/services/hdinsight/mgmt/2015-03-01-preview/hdinsight/models.go#L1368)

Reply:
Yes 
o Azure Datalake store is configure throught  configuration section with core-site and cluster_identity properties